app/src/main/java/com/lorenzovainigli/foodexpirationdates/viewmodel/ExpirationDatesViewModel.kt

```diff
@@ -17,9 +21,15 @@ class ExpirationDatesViewModel @Inject constructor(
     private var expirationDates : Flow<List<ExpirationDate>> = flowOf(emptyList())
     private var expirationDate : ExpirationDate? = null
 
+    private val _isSplashScreenLoading: MutableState<Boolean> = mutableStateOf(value = true)
+    val isSplashScreenLoading: State<Boolean> = _isSplashScreenLoading
+
     fun getDates(): Flow<List<ExpirationDate>> {
         viewModelScope.launch {
+            _isSplashScreenLoading.value = true
             expirationDates = repository.getAll()
+            delay(2000)
+            _isSplashScreenLoading.value = false
```

Comment:
Good. This codes added an additional delay to the call `repository.getAll()`, so I think it's better to use the delay to execute the call to the database. I propose this code:

```kotlin
val deferred = async {
    expirationDates = repository.getAll()
}
delay(1000)
deferred.await()
```
Here the call to the database is executed asynchronously, and the delay don't wait the db call to terminate. However, if the call to the db is longer that the delay, the `await()` tells the app to don't visualize the list of item unit the db call returned.
Also I think 1 second delay is enough.
Do you agree?

Reply:
awesome, I have updated the code as your great advice. strongly agreed. 

tested device for splash screen area (asset not overlapped) :
redmi 9 A12 ✅ | emulator Pixel 5 A11 ✅ | other devices ❎